### PR TITLE
PCQ_694 - Added Exception and Feign classes

### PR DESCRIPTION
**Pull request by ManojKulkarni-HMCTS**
### JIRA link (if applicable) ###
PCQ-694

### Change description ###
Added more common classes. Please refer to the description in PCQ-694 for details.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/pcq/commons/exception/ExternalApiExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+public class ExternalApiExceptionTest {
+
+    @Test
+    public void externalApiExceptionTest() {
+        ExternalApiException externalApiException = new ExternalApiException(BAD_REQUEST, "BAD REQUEST");
+
+        assertEquals("`Not expected status", "400 BAD_REQUEST", externalApiException.getHttpStatus().toString());
```
Comment (davejones74) [inline src/test/java/uk/gov/hmcts/reform/pcq/commons/exception/ExternalApiExceptionTest.java:14]: Funny  character in assert message?

File: src/main/java/uk/gov/hmcts/reform/pcq/commons/config/FeignInterceptorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Enumeration;
+import java.util.Locale;
+
+
+@Slf4j
+@SuppressWarnings("PMD.ConfusingTernary")
```
Comment (davejones74) [inline src/main/java/uk/gov/hmcts/reform/pcq/commons/config/FeignInterceptorConfiguration.java:15]: Does ConfusingTernary still apply?

File: src/test/java/uk/gov/hmcts/reform/pcq/commons/utils/JsonFeignResponseUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    @Test
+    public void test_toResponseEntity_with_payload_not_empty() {
+        Map<String, Collection<String>> header = new ConcurrentHashMap<>();
+        Collection<String> list = new ArrayList<>(Arrays.asList("a", "b"));
+        header.put(ENCODING_STR, list);
+
+        Response response = Response.builder().status(200).reason("OK").headers(header).body(
+                "{"
```
Comment (davejones74) [inline src/test/java/uk/gov/hmcts/reform/pcq/commons/utils/JsonFeignResponseUtilTest.java:145]: This JSON looks the same as previous line 43, also if this isn't coming from a file can it be made a static final string?

Comment (ManojKulkarni-HMCTS) [reply src/test/java/uk/gov/hmcts/reform/pcq/commons/utils/JsonFeignResponseUtilTest.java]: Agreed. Created a static final string.

File: src/main/java/uk/gov/hmcts/reform/pcq/commons/config/FeignInterceptorConfiguration.java (hunk 2 above)
Comment (ManojKulkarni-HMCTS) [reply src/main/java/uk/gov/hmcts/reform/pcq/commons/config/FeignInterceptorConfiguration.java]: Removed the annotation

File: src/test/java/uk/gov/hmcts/reform/pcq/commons/exception/ExternalApiExceptionTest.java (hunk 1 above)
Comment (ManojKulkarni-HMCTS) [reply src/test/java/uk/gov/hmcts/reform/pcq/commons/exception/ExternalApiExceptionTest.java]: Removed the funny character

**Review by davejones74 — APPROVED** [review]

**Pull request merged**